bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,881 @@
+        if (localWebSocketClient == null) {
+            websocketClient = new WebSocketClient(httpClient);
+
+            localWebSocketClient = websocketClient;
+
+            if (localWebSocketClient != null) {
+                localWebSocketClient.setExecutor(jettyThreadPool);
+
+                socketMonitor.start();
+
+                ret = true;
+            } else {
+                ret = false;
+            }
+        } else {
+            ret = true;
+        }
+
+        return ret;
+    }
+
+    /**
+     * Method to initialize the bridge
+     */
+    @Override
+    public void initialize() {
+        httpConnectionOK.set(false);
+
+        // load configuration
+        FreeAtHomeBridgeHandlerConfiguration locConfig = getConfigAs(FreeAtHomeBridgeHandlerConfiguration.class);
+
+        ipAddress = locConfig.ipAddress;
+        if (ipAddress.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/conf-error.ip-address-missing");
+            return;
+        }
+
+        password = locConfig.password;
+        if (password.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/conf-error.password-missing");
+            return;
+        }
+
+        username = locConfig.username;
+        if (username.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/conf-error.username-missing");
+            return;
+        }
+
+        // build base URL
+        baseUrl = "http://" + ipAddress + "/fhapi/v1/api";
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(() -> {
+            boolean thingReachable = true;
+
+            // Open Http connection
+            if (!openHttpConnection()) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/comm-error.http-wrongpass-or-ip");
+
+                thingReachable = false;
+            }
+
+            // Open the websocket connection for immediate status updates
+            if (!openWebSocketConnection()) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "@text/comm-error.not-able-open-websocketconnection");
+
+                thingReachable = false;
+            }
+
+            if (thingReachable) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+        });
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        super.handleConfigurationUpdate(configurationParameters);
+
+        // load configuration
+        FreeAtHomeBridgeHandlerConfiguration locConfig = getConfigAs(FreeAtHomeBridgeHandlerConfiguration.class);
+
+        ipAddress = locConfig.ipAddress;
+        password = locConfig.password;
+        username = locConfig.username;
```

Comment:
I'm not sure if you need to override this method at all, or if it would be sufficient to let `initialize()` handle configuration updates? It's not enough to just change these strings anyway, since a change in IP address, for example, would require you to reconnect also. Validation is also missing here (like implemented in `initialize`).